### PR TITLE
Snowpark blog entry

### PR DESCRIPTION
**Pull request by zzeekk**
Blog post about migrating getting-started data pipeline to Snowflake/Snowpark.

### CODE REVIEW

File: blog/2022-04-06-SDL-snowpark/2022-04-06-SDL-snowpark.md [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+Snowpark is good for transforming data inside Snowflake, but not all data might be located in Snowflake and suitable for Snowflake. 
+Here it is interesting to use Spark and its many connectors, in particular to ingest and export data.
+
+Combining Spark and Snowpark in a smart data pipeline using a DataFrame API would be the ideal solution.
+With the integration of Snowpark as engine in SDLB we created just that. 
+
+This blog post will show how to migrate our example data pipeline of the [Getting Started](../../docs/getting-started/setup) guide Part 1 to use Spark for ingestion and Snowpark for transformation.
+
+## Prerequisits
```
Comment (Geheiner) [inline blog/2022-04-06-SDL-snowpark/2022-04-06-SDL-snowpark.md]: I'm mostly fixing some typos.
Did you know that IntelliJ does spell checking?
image

File: blog/2022-04-06-SDL-snowpark/2022-04-06-SDL-snowpark.md [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+      }
+
+Note that the attribute `db` of the SnowflakeTableDataObject should be filled with the schema of the Snowflake table and that this is *not* the same as the attribute `database` of SnowflakeTableConnection. 
+
+## Migrating Actions
+
+The new SDLB version introduced some naming changes:
+- The CustomSparkAction can now also process Snowpark-DataFrames and is therefore renamed to CustomDataFrameAction.
+- The ScalaClassDfTransformer was specific for Spark. In the new SDLB version there is a specific scala-class DataFrame transformer for Spark and Snowpark, e.g. ScalaClassSparkDfTransformer and ScalaClassSnowparkDfTransformer. And there is even a ScalaClassGenericDfTransformer to implement transformations using a unified API. In our case we will migrate the transformation to use Snowpark and set the type to ScalaClassSnowparkDfTransformer.
+- See [Architecture](../../docs/architecture) chapter "breaking changes" for other renamings.
```
Comment (Geheiner) [inline blog/2022-04-06-SDL-snowpark/2022-04-06-SDL-snowpark.md]: does not exist yet, to be added later

**Review by Geheiner — APPROVED** [review]
Nice blog post! I read it and found it easy to follow. I did not have time to run it myself, though.

File: docs/reference/executionEngines.md [hunk 3]
```diff
@@ -4,16 +4,18 @@ title: Execution Engines
 ---
 
 An execution engine is a technology/library used by SDLB to transform data. SDLB supports different execution engines and is able to combine different execution engines in the same data pipeline / job.
-Execution engines can be categorized based on the data structure they provide transport and transform data. The transport object of data between DataObjects and Actions is called a SubFeed.
+The data structure used to transport data between DataObjects and Actions is called a SubFeed.
```
Comment (Geheiner) [inline docs/reference/executionEngines.md:7]: FYI I hope you can live with that new description. A bit less precise, but easier to understand

File: docs/reference/executionEngines.md [hunk 4]
```diff
@@ -36,3 +36,16 @@ A "Generic DataFrame API" Action can run with different execution engines like S
 To check which execution engine was chosen, look for logs like the following:
 
       INFO  CustomDataFrameAction - (Action~...) selected subFeedType SparkSubFeed
+
+### Execution Engines vs Execution Environments
```
Comment (Geheiner) [inline docs/reference/executionEngines.md:40]: Not 100% sure if everything is correct, feel free to edit/remove it if you want to update it later

File: blog/2022-04-06-SDL-snowpark/2022-04-06-SDL-snowpark.md (hunk 1 above)
Comment (zzeekk) [reply blog/2022-04-06-SDL-snowpark/2022-04-06-SDL-snowpark.md]: Mh, i was working in visual studio code, but you're right, i should look for a spell checking plugin.

File: blog/2022-04-06-SDL-snowpark/2022-04-06-SDL-snowpark.md (hunk 2 above)
Comment (zzeekk) [reply blog/2022-04-06-SDL-snowpark/2022-04-06-SDL-snowpark.md]: I remove it. It should point to github release notes but there the breaking changes for the current Snapshot-Version are not yet released...

File: docs/reference/executionEngines.md (hunk 4 above)
Comment (zzeekk) [reply docs/reference/executionEngines.md:40]: Thanks, did some rephrasing.

**Pull request merged**